### PR TITLE
Gameplay package initial tests.

### PR DESCRIPTION
**Pull request by MettyS**
Initial basic move validation and implementation.

### CODE REVIEW

File: server/gameplay/game_hander_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+	return map[string]d.GameRole{
+		"player1": d.PlayerWhite,
+		"player2": d.PlayerBlack,
+		"player3": d.Spectator,
+	}
+}
+
+func gameWithPlayersFixture() Game {
+	game := Game{}
+	game.Board = []d.Tile{2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}
```
Comment (jer-irl) [inline server/gameplay/game_hander_test.go]: I'd prefer a little more of a human-readable test input here :)

File: server/gameplay/game_hander_test.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+			indexFrom = uint32(num)
+		} else {
+			indexTo = uint32(num)
+			moves = append(moves, d.Move{IndexFrom: indexFrom, IndexTo: indexTo})
+		}
+	}
+	return moves
+}
+
+type (
```
Comment (jer-irl) [inline server/gameplay/game_hander_test.go]: Can you change this to just `type name whatever` without the multiple lines and parens?

**Review by jer-irl — CHANGES_REQUESTED** [review]

File: server/gameplay/game_hander_test.go (hunk 1 above)
Comment (MettyS) [reply server/gameplay/game_hander_test.go]: Updated this!

File: server/gameplay/game_hander_test.go (hunk 2 above)
Comment (MettyS) [reply server/gameplay/game_hander_test.go]: Thanks for catching this.

File: server/gameplay/game_hander_test.go [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+               ww..ww..ww..ww..`)
+	game.Board = createBoardFromString(board)
+	game.CurrentPlayer = d.PlayerWhite
+	game.Participants = playersFixture()
+	return game
+}
+
+func gameWithBlackTurn() Game {
+	game := Game{}
+	board := (`..bb..bb..bb..bb
```
Comment (jer-irl) [inline server/gameplay/game_hander_test.go]: Canyon appease the formatter as well as my eyes trying to read the offsets here and above?

File: server/gameplay/game_hander_test.go [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+			result += fmt.Sprintf("..%v", tileString)
+		} else {
+			result += fmt.Sprintf("%v..", tileString)
+		}
+	}
+	result += fmt.Sprintf("\n")
+	return result
+}
+
+func printHorizontalRule() {
```
Comment (jer-irl) [inline server/gameplay/game_hander_test.go]: unused function

**Review by jer-irl — APPROVED** [review]

File: server/gameplay/game_hander_test.go (hunk 4 above)
Comment (MettyS) [reply server/gameplay/game_hander_test.go]: Removed this, thanks.

**Pull request merged**